numaprom/udf/inference.py

```diff
@@ -49,18 +51,26 @@ def inference(_: str, datum: Datum) -> List[Tuple[str, bytes]]:
     metric_config = get_metric_config(payload.composite_keys["name"])
     model_config = metric_config["model_config"]
 
-    artifact_data = load_model(
-        skeys=[payload.composite_keys["namespace"], payload.composite_keys["name"]],
-        dkeys=[model_config["model_name"]],
-    )
-
     train_payload = {
         "uuid": payload.uuid,
         **payload.composite_keys,
         "model_config": model_config["name"],
         "resume_training": False,
     }
 
+    if payload.status == Status.ARTIFACT_NOT_FOUND:
+        _LOGGER.info(
```

Comment:
How are we differentiating the type of training? 

are we training both preprocessing models and main models all the times?

Reply:
Yes all times. Doesn't matter which model is missing. if any model is missing, we need to retrain.